File: executable/executable.py
#!/usr/bin/python
import os, sys
sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))) + "/build") # Prefer local library over installed one
import devtemplate

if __name__ == "__main__":
    d = devtemplate.Devtemplate()
    correct = d.devtemplate() == 42
    print("Devtemplate is functioning correctly" if correct else "Devtemplate is malfunctioning")
    sys.exit(0 if correct else 1)

File: executable/executable.cpp
#include <devtemplate/devtemplate.h>
#include <iostream>

int main()
{
    devtemplate::Devtemplate d;
    #ifndef DEVTEMPLATE_ADVANCED
        const bool correct = d.calculate() == 42;
    #else
        const bool correct = d.calculate_advanced() == 43;
    #endif
    std::cout << devtemplate::Devtemplate::version() << (correct ? " is functioning correctly" : " is malfunctioning") << std::endl;
    return correct ? 0 : 1;
}